File: day01/star02.py
def parse_line(line):
    print('-'*10 + '\n' + line)  #DELME
    found = True
    while found:
        found = False
        for i in range(len(line)):
            for j, number in enumerate(['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']):
                if line[i:i+len(number)] == number:
                    line = line[:i] + str(j+1) + line[i+len(number):]
                    # line = line[:i].replace(number, str(j+1), 1)
                    found = True

    print(line)  #DELME
    digits = [c for c in line if c in '0123456789']
    print(digits)
    number = int(digits[0])*10 + int(digits[-1])
    print(number)
    return number

sum = 0
with open('input.txt') as f:
    i = 0  #DELME
    for line in f:
        sum += parse_line(line)
        i += 1
        # if i>10:
        #     break

print(sum)

File: day01/star02.rs
use std::fs;

fn parse_line(line: &str) -> u32 {
    let mut digits = Vec::new();
    for (i, c) in line.chars().enumerate() {
        if c.is_digit(10) {
            digits.push(c.to_digit(10).unwrap());
        } else if line[i..].starts_with("one") {
            digits.push(1);
        } else if line[i..].starts_with("two") {
            digits.push(2);
        } else if line[i..].starts_with("three") {
            digits.push(3);
        } else if line[i..].starts_with("four") {
            digits.push(4);
        } else if line[i..].starts_with("five") {
            digits.push(5);
        } else if line[i..].starts_with("six") {
            digits.push(6);
        } else if line[i..].starts_with("seven") {
            digits.push(7);
        } else if line[i..].starts_with("eight") {
            digits.push(8);
        } else if line[i..].starts_with("nine") {
            digits.push(9);
        }
    }
    return 10*digits[0] + digits[digits.len()-1];
}

fn main() {
    let contents = fs::read_to_string("input.txt")
    .expect("Should have been able to read the file");

    let mut sum: u32 = 0;
    for (i, line) in contents.split('\n').enumerate() {
        sum += parse_line(line);
    }
    println!("{sum}");
}